ekan0ra/bot.py

```diff
@@ -0,0 +1,448 @@
+                    application_logger.error(
+                        'Error adding admin!', exc_info=True
+                    )
+
+            elif msg.startswith('.rm'):
+                try:
+                    nick = msg.split()[1]
+                    self.remove_admin(nick)
+                    self.say(self.channel, '%s removed from admin.' % nick)
+                    application_logger.info('%s removed from admin.', nick)
+                except Exception, err:
+                    application_logger.error(
+                        'Error removing admin!', exc_info=True
+                    )
+
+            elif msg.lower().startswith('.pingall'):
+                self.pingmsg = msg.lower().lstrip('.pingall')
+                self.names(channel).addCallback(self.pingall)
+        # end processing admin message
+
+
+        # User wants to ask a question
+        if msg == '!'  and self.islogging:
+            self.qs_queue.enqueue(user.nick)
+            self.describe(
+                self.channel, 'Queue: [%r]' % ', '.join(self.qs_queue))
+
+        # User no longer wants to ask a question; remove them from queue
+        elif msg in ('!-', '!!')  and self.islogging:
+            self.qs_queue.dequeue(user.nick)
+            self.describe(
+                self.channel, 'Queue: [%r]' % ', '.join(self.qs_queue))
+
+        elif msg in ('!', '!-', '!!') and not self.islogging:
+            self.say(
+                self.channel,
+                '%s: No session is going on, feel free to ask a question. '
+                    'You do not have to type %s' % (user.nick, msg)
+            )
+        # end processing question indicator   
+
+        elif msg == '.givemelogs':
+            if not self.last_log_filename:
+                self.say(user.nick, 'Sorry, I do not have the last log.')
+                application_logger.warning('Could not find last class log!')
+            else:
+                sys.argv = ['fpaste', self.last_log_filename]
+                try:
+                    short_url, url = fpaste.main()
+                    application_logger.info(
+                        'Class Log uploaded; and Fedora Paste returned:\n\t'
+                            'Short URL: %s\n\tLong URL: %s', short_url, url
+                    )
+                    self.say(user.nick, url)
+                except:
+                    self.say(user.nick, 'Hit a 500! I have a crash on you.')
+                    application_logger.error(
+                        'Log uploading to Fedora failed!', exc_info=True
+                    )
+        
+        elif msg == '.help':
+            for command, help_txt in commands.iteritems():
+                self.say(
+                    user.nick,
+                    help_template.format(
+                        command=command, help_text=help_txt
+                    )
+                )      
+
+        elif msg.startswith('.link'):
+            self.links_for_key(msg)
```

Comment:
Do you mean, have the `.link` command to be usable only by an admin or its argument `-l` only available to an admin?